syntax = "proto3";

package TestPackage;

service TestStream
{
    rpc TestSimple(TestRequest) returns (TestResponse);
    rpc TestClientStream(stream TestRequest) returns (TestResponse);
    rpc TestServerStream(TestRequest) returns (stream TestResponse);
    rpc TestDoubleStream(stream TestRequest) returns (stream TestResponse);

    rpc ServerStreamApple(stream TestRequest) returns (stream TestResponse);
    rpc ServerStreamPear( TestRequest) returns (stream TestResponse);
    rpc ServerStreamMango( TestRequest) returns (stream TestResponse);

    rpc DoubleStreamApple(stream TestRequest) returns (stream TestResponse);
    rpc DoubleStreamPear(stream TestRequest) returns (stream TestResponse);
    rpc DoubleStreamMango(stream TestRequest) returns (stream TestResponse);    
};

message TestRequest
{
    string name = 1;
    string time = 2;
    string session_id = 3;
    string obj_id = 4;
};


message TestResponse
{
    string name = 1;
    string time = 2;
    string session_id = 3;
    string obj_id = 4;
};